src/constructors/kneser.c

```diff
@@ -0,0 +1,65 @@
+/*
+   IGraph library.
+   Copyright (C) 2022  The igraph development team
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+*/
+
+#include "igraph_vector_list.h"
+
+static igraph_integer_t igraph_i_number_of_subsets(igraph_integer_t n, igraph_integer_t k){
+    igraph_integer_t r = (k < n - k ? k : n - k);
+    igraph_integer_t number_of_subsets = 1;
+
+    for(igraph_integer_t i = 1 ; i <= r ; i++) {
+        number_of_subsets *= n - i + 1;
+        number_of_subsets /= i;
+    }
+
+    return number_of_subsets;
+}
+
+static igraph_error_t igraph_i_generate_subsets(igraph_integer_t n, igraph_integer_t k,
+                              igraph_integer_t subsets, igraph_vector_int_list_t *vertices) {
+
+    igraph_vector_int_t *temp = igraph_vector_list_get_ptr(vertices, 0);
+    IGRAPH_CHECK(igraph_vector_resize(temp, k));
```

Comment:
Since `temp` is an `igraph_vector_int_t`, you need to use the functions that take an `igraph_vector_int_t`. For instance, `igraph_vector_int_resize()` instead of `igraph_vector_resize()`, `igraph_vector_int_update()` instead of `igraph_vector_update()` and so on. I haven't checked the correctness of the algorithm yet, this just simply struck my eye.

I haven't checked the correctness of this function, this just simply struck my eye.

Reply:
Oh, I must have missed this. I'll fix it.